st2common/st2common/models/api/action.py

```diff
@@ -148,6 +149,11 @@ class ActionAPI(BaseAPI):
                 "patternProperties": {
                     "^\w+$": util_schema.get_draft_schema()
                 }
+            },
+            "tags": {
+                "description": "User associated metadata assigned to this object.",
```

Comment:
To clarify it - I'm against complex values and +1 for keeping it the way it is now - only supporting string values for both, key/name and the value where.

The value is an opaque blob to us anyway so user can encode and store arbitrary stuff there.

On a related note - we should limit the maximum field size to prevent resource exhaustion, ddos and similar attacks.

This is something which I don't think we currently do, but we should do for all the string fields. That's of course not blocking this PR and just something we should do soon in a different PR.